### ISSUE TITLE
twisted.mail.imap4.encoder and twisted.mail.imap4.decoder have incorrect signature

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2929]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#2929|
|Type|enhancement|
|Created|2007-11-30 00:24:23Z|
|Branch|https://github.com/twisted/twisted/tree/encoder-decoder-signature-2929|

These two functions are supposed to have the same signature as Codec.encode and Codec.decode.  Instead, they each take a single argument.  They should be changed to have a second optional argument, _errors_.

They probably won't implement anything special for any errors policy, though.

<details><summary>Searchable metadata</summary>

```
trac-id__2929 2929
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__branches_encoder_decoder_signature_2929 branches/encoder-decoder-signature-2929
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1196382263000000 1196382263000000
changetime__1197086634000000 1197086634000000
version__None None
owner__ 
cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [22010]) Branching to 'encoder-decoder-signature-2929'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Ready for review

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [22033]) Merge encoder-decoder-signature-2929

Author: exarkun
Reviewer: therve
Fixes [#2929](https://github.com/twisted/twisted/issues/2929)

Change the signatures of twisted.mail.imap4.encoder and twistedmail.imap4.decoder
so that they accept a second argument defining the error handling behavior to use.
This makes them compatible with the documentation for codec encoder and decoder
functions and makes the IMAP4-UTF-7 codec registered by the imap4 module work
with the two-argument form of str.encode and str.decode.

The error behavior parameter is currently ignored.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
